services/QuillLMS/app/models/topic.rb

```diff
@@ -51,6 +51,19 @@ def level_three?
     level == 3
   end
 
+  def activity_count
+    child_topics = Topic.where(visible: true, parent_id: id)
+    case level
+    when 3
+      grandchild_topics = child_topics.map { |ct| Topic.where(parent_id: ct.id) }.flatten
+      grandchild_topics.map { |gct| gct.activities.count}.reduce(:+)
+    when 2
+      child_topics.map { |gct| gct.activities.count}.reduce(:+)
```

Comment:
Ditto about using `sum`.

Also, the loop variable should probably be `ct` instead of `gct`.